wgpu-core/src/error.rs

```diff
@@ -0,0 +1,175 @@
+use core::fmt;
+use std::error::Error;
+
+use crate::{
+    gfx_select,
+    hub::{Global, IdentityManagerFactory},
+};
+
+pub struct ErrorFormatter<'a> {
+    writer: &'a mut dyn fmt::Write,
+    global: &'a Global<IdentityManagerFactory>,
+}
+
+impl<'a> ErrorFormatter<'a> {
+    pub fn error(&mut self, err: &dyn Error) {
+        writeln!(self.writer, "    {}", err).expect("Error formatting error");
+    }
+
+    pub fn note(&mut self, note: &dyn fmt::Display) {
+        writeln!(self.writer, "      note: {}", note).expect("Error formatting error");
```

Comment:
any reason to not use "\t" here?

Reply:
Are we going to have _that_ discussion, tabs vs spaces? ;)
I didn't really intend functional changes here, and that's how it was. I feel the tabs would indent a bit too much though.

For reference, `anyhow` seems to be doing [indenting with spaces](https://github.com/dtolnay/anyhow/blob/master/src/fmt.rs#L73)